components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OAuthUserConsentedScopesDAO.java

```diff
@@ -0,0 +1,78 @@
+package org.wso2.carbon.identity.oauth2.dao;
+
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2ScopeConsentException;
+import org.wso2.carbon.identity.oauth2.model.UserApplicationScopeConsentDO;
+
+import java.util.List;
+
+/**
+ * This interface defines the API for user consent management for OAuth scopes.
+ */
+public interface OAuthUserConsentedScopesDAO {
+
+    /**
+     * Retrieve the user consent given for OAuth scopes for a given application.
+     *
+     * @param userId    User identifier.
+     * @param appId     Application identifier.
+     * @param tenantId  Tenant Id.
```

Comment:
is this the application tenant id or user tenant id?

Reply:
This will the user tenant ID.